File: Encryption/decrypt.py
def decryptLetter(letter, multKey, addKey):
    if not letter.isalpha():
        return letter
    num = ord(letter)-65
    num = ((num + addKey) * multKey) % 26
    return chr(num+65)


if __name__ == '__main__':
    message = input("Enter your message:\t\t").upper()
    multKey = int(input("Enter your multiplicative key:\t"))
    addKey = int(input("Enter your additive key:\t"))

    invList = [1,0,9,0,21,0,15,0,3,0,19,0,0,0,7,0,23,0,11,0,5,0,17,0,25];
    multInv = invList[multKey-1];
    addInv = 26 - addKey;

    secret_message = ""
    for c in message:
        secret_message += decryptLetter(c, multInv, addInv)
    print("\nYour decrypted message is\n\n" + secret_message)

File: Encryption/decrypt.c
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>
#include <errno.h>
#include <ctype.h>

void print_usage(char *prog_name) {
  printf("correct usage: %s <multiplicative key> <additive key>", prog_name);
  printf(" <src filename> <dest filename>\n\n");
}

int main(int argc, char** argv) {
  if (argc != 5) {
    print_usage(argv[0]);
    return -1;
  }
  // parse args
  int invList[] = {1,0,9,0,21,0,15,0,3,0,19,0,0,0,7,0,23,0,11,0,5,0,17,0,25};
  int multKey = atoi(argv[1]);
  int multInv = invList[multKey-1];
  int addKey = atoi(argv[2]);
  // open files for I/O
  FILE *cipherF = fopen(argv[3], "r");
  if (cipherF == NULL) {
    perror(NULL);
    return -1;
  }
  FILE *plainF = fopen(argv[4], "w+");
  // compute cipher
  char c;
  while ((c = getc(cipherF)) != EOF) {
    // compute cipher
    if (isalpha(c)) {
      int i = toupper(c) - 65 + 26;
      i = ((i - addKey) * multInv) % 26;
      c = (char) (i + 65);
    }
    putc(c, plainF);
  }
  // close files
  fclose(cipherF);
  fclose(plainF);
  return 0;
}
